src/components/Planning/GroupForm/GroupForm.tsx

```diff
@@ -0,0 +1,333 @@
+  }
+  payload.projects = projects.length > 0 ? projects.map((p) => p.value) : [];
+
+  return payload;
+};
+
+const GroupForm: FC = () => {
+  const masterClasses = useAppSelector(selectMasterClasses);
+  const classes = useAppSelector(selectClasses);
+  const subClasses = useAppSelector(selectSubClasses);
+  const districts = useAppSelector(selectDistricts);
+  const divisions = useAppSelector(selectDivisions);
+  const subDivisions = useAppSelector(selectSubDivisions);
+
+  const getReverseLocationHierarchy = useCallback(
+    (subDivisionId: string | undefined) => {
+      const classAsListItem = (projectLocation: ILocation | undefined): IListItem => ({
+        id: projectLocation?.id || '',
+        value: projectLocation?.name || '',
+      });
+
+      const selectedSubDivision = subDivisions.find((sd) => sd.id === subDivisionId);
+
+      const selectedDivision = divisions.find((d) => d.id === selectedSubDivision?.parent);
+
+      const selectedDistrict = districts.find(
+        (D) => D.id === selectedDivision?.parent && D.parent === null,
+      );
+      return {
+        division: listItemToOption(classAsListItem(selectedDivision) || []),
+        subDivision: listItemToOption(classAsListItem(selectedSubDivision) || []),
+        district: listItemToOption(classAsListItem(selectedDistrict) || []),
+      };
+    },
+    [divisions, districts, subDivisions],
+  );
+
+  const getReverseClassHierarchy = useCallback(
+    (subClassId: string | undefined) => {
+      const classAsListItem = (projectClass: IClass | undefined): IListItem => ({
+        id: projectClass?.id || '',
+        value: projectClass?.name || '',
+      });
+
+      const selectedSubClass = subClasses.find((sc) => sc.id === subClassId);
+
+      const selectedClass = classes.find((c) => c.id === selectedSubClass?.parent);
+
+      const selectedMasterClass = masterClasses.find(
+        (mc) => mc.id === selectedClass?.parent && mc.parent === null,
+      );
+      return {
+        _class: listItemToOption(classAsListItem(selectedClass) || []),
+        subClass: listItemToOption(classAsListItem(selectedSubClass) || []),
+        masterClass: listItemToOption(classAsListItem(selectedMasterClass) || []),
+      };
+    },
+    [classes, masterClasses, subClasses],
+  );
+
+  const { formMethods, formValues, formFields } = useGroupForm();
+  const {
+    handleSubmit,
+    reset,
+    formState: { dirtyFields, isDirty },
+    watch,
+    setValue,
+    getValues,
+  } = formMethods;
+
+  const dispatch = useAppDispatch();
+  const { Header, Content, ActionButtons } = Dialog;
+  const { t } = useTranslation();
+
+  const [formState, setFormState] = useState<IFormState>({
+    isOpen: false,
+    selectedClass: '',
+    projectsForSubmit: [],
+    selectedLocation: '',
+    showAdvanceFields: false,
+  });
+
+  const { isOpen, selectedClass, selectedLocation, projectsForSubmit, showAdvanceFields } =
+    formState;
+
+  useClassList(true, selectedClass);
```

Comment:
useClassList and useLocationList could be called inside the useGroupForm-hook.

Reply:
Now use new hook